ocaml/xapi/xapi_pif.ml

```diff
@@ -105,6 +105,39 @@ let read_bridges_from_inventory () =
   with _ ->
     []
 
+(* Ensure the PIF is not a bond slave. *)
+let assert_not_bond_slave_of ~__context ~self =
+  if Db.PIF.get_bond_slave_of ~__context ~self <> Ref.null then
+    raise (Api_errors.Server_error (Api_errors.cannot_plug_bond_slave, [Ref.string_of self]))
+
+(* Ensure the PIF has valid IPv4 configuration mode. *)
+let assert_valid_ip_configuration ~__context ~self =
+  if Db.PIF.get_ip_configuration_mode ~__context ~self = `None then
+    raise (Api_errors.Server_error(Api_errors.pif_has_no_network_configuration, [Ref.string_of self]))
+
+(* Ensure the PIF has valid IPv6 configuration mode. *)
+let assert_valid_ipv6_configuration ~__context ~self =
+  if Db.PIF.get_ipv6_configuration_mode ~__context ~self = `None then
+    raise (Api_errors.Server_error(Api_errors.pif_has_no_v6_network_configuration, [Ref.string_of self]))
+
+let assert_usable_for_management ~__context ~for_mgmt ~required_address_type ~self =
+  assert_not_bond_slave_of ~__context ~self;
+  let primary_address_type = Db.PIF.get_primary_address_type ~__context ~self in
+  if for_mgmt && required_address_type <> primary_address_type then
+    raise (Api_errors.Server_error(Api_errors.pif_incompatible_primary_address_type, [Ref.string_of self]));
+  if Db.PIF.get_managed ~__context ~self = true then begin
+    match primary_address_type with
+      | `IPv4 -> assert_valid_ip_configuration ~__context ~self 
+      | `IPv6 -> assert_valid_ipv6_configuration ~__context ~self 
+      | _ -> try
+        let mgmt_pif = Helpers.get_management_interface ~__context ~host:(Helpers.get_localhost ~__context) in
```

Comment:
This does not look right to me. In the original code, this block use to be always executed in case there were no errors, but now only if the `primary_address_type` of the management-PIF-to-be is invalid. The point of this check is that a call to `{host;pool}_management_reconfigure` does not switch the host/pool from IPv4 to IPv6 or vice versa.